dataset_interfaces/interface.py

```diff
@@ -93,6 +103,10 @@ def get_path(self, run_name: str) -> Path:
         return make_testdef_path(run_name, self.dataset_name, self.example_id)
 
     def __post_init__(self):
+        assert self.dataset_generator is not None
+        if self.evaluation_fn is None:
+            self.evaluation_fn = self.dataset_generator.evaluate_correct
```

Comment:
If you are doing some `TestExample` attributes as properties, you can also add this/